changelogs/3.1.0.md

```diff
@@ -0,0 +1,253 @@
+  `net_box.connect` to yield, despite being required to be fully asynchronous
+  (gh-9489).
+* Fixed a bug in `net.box` when a connection with asynchronous requests could
+  get garbage collected (gh-9629).
+* Fixed an inconsistency between the documented `on_disconnect` trigger behavior
+  of `net.box` connections when an error is thrown and the actual behavior
+  (gh-9717).
+* Fixed a bug in the `on_disconnect` trigger of `net.box` connections that
+  caused Tarantool server to hang indefinitely when an error was thrown from the
+  trigger (gh-9797).
+* The `on_schema_reload` trigger behavior of `net.box` connections when an
+  error is thrown is now consistent with the behavior of the `on_disconnect`
+  trigger (gh-9679).
+* Fixed a bug when a `net.box` connection remained active after being closed
+  from the connection's `on_connect` trigger (gh-9827).
+
+### Box
+
+* Now all triggers either have a direct impact on the execution flow (for
+  example, closing a connection or throwing an error) or print a message to
+  the error log when they throw an error (gh-9309).
+* The `exclude_null` option is now supported by functional indexes (gh-9732).
+* Provided old and new tuples in `box.error.TUPLE_FOUND` (gh-6166).
+* An attempt to execute a DDL, DML, DQL, DCL or TCL query within
+  a transactional trigger (`on_commit` or `on_rollback`) now fails
+  with a specific error (gh-9186, gh-7331).
+* Fixed a memory leak when dropping fully-temporary spaces (gh-9296).
+
+### Config
+
+* Fixed an issue when a leader fails to start with the `attempt to index a nil
+  value` error if a config with all UUIDs set is used during a cluster's
+  bootstrap (gh-9572).
+* Fixed a bug when only one alert of a particular type is reported in
+  `config:info().alerts` instead of several ones. Examples of such alerts are
+  privilege grant delay due to a lack of a space/function/sequence and skipping
+  of a non-dynamic `box.cfg()` option applies on the configuration reloading
+  (gh-9586).
+* An alert regarding delayed privilege granting is now cleared when the
+  privilege is granted (gh-9574).
+* Calling `require('config'):get()` in role code now returns the configuration
+  that is currently applied (gh-9649).
+* The config status is now set after the `post_apply` phase (ghe-643).
+* A non-existent role can now be assigned in the `credential` section of
+  the configuration (gh-9643).
+* Privileges that were not granted by the config module are no longer
+  revoked by the config module (gh-9643).
+* Fixed a non-verbose error on an empty configuration file.
+  Now Tarantool can successfully start up with an empty configuration
+  file using data from other configuration sources (gh-9845).
+* Added additional validation to a cluster's configuration.
+  Now it is forbidden to create an empty group or
+  replicaset (gh-9895).
+
+### Datetime
+
+* Fixed a bug in `strptime` when the assertion was triggered (gh-8525).
+
+### Httpc
```

Comment:
```suggestion
### Http client
```

For consistency with the **Functionality added** section (see line 72)

Reply:
done.